src/go/k8s/controllers/redpanda/cluster_controller.go

```diff
@@ -44,10 +45,11 @@ var (
 // ClusterReconciler reconciles a Cluster object
 type ClusterReconciler struct {
 	client.Client
-	Log                  logr.Logger
-	configuratorSettings resources.ConfiguratorSettings
-	clusterDomain        string
-	Scheme               *runtime.Scheme
+	Log                   logr.Logger
+	configuratorSettings  resources.ConfiguratorSettings
+	clusterDomain         string
+	Scheme                *runtime.Scheme
+	AdminAPIClientFactory utils.AdminAPIClientFactory
```

Comment:
Good question.. In general it's because the controller is singleton, while the admin API client depends on the cluster to contact. Anyway, I could have put the client directly without using the factory, since I inject a different one only in tests, but it didn't make a lot of sense outside the testing scope..

Reply:
I thought it was a little confusing that it's checked if `AdminAPIClientFactory` is `nil`, and if it is, then NewAdminAPI is called directly. It's an unexpected side effect in my opinion.

Reply:
Yeah, agreed. Better setting it always.